### PR TITLE
Update: More share improvemnts

### PR DESCRIPTION
**Pull request by adodge-bcg**
# Description

This PR includes the following proposed change(s):

- updated share language to be spaced similar to rest of page
- updated buttons displayed when share is available
- changed button height to min-hight for longer button names on small screens
- added error handling for non-400 errors
- updated tests/snapshots to reflect button/text changes
- added test for share flow, and error handling

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Tested locally, in storybook, passing existing and new Jest tests.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/ecrc-frontend/src/components/page/informationreview/InformationReview.js [hunk 1]
```diff
@@ -100,9 +100,25 @@ export default function InformationReview({
...
-          // If error = no share, do nothing
-          // else if other error, go to error page and display correct error
+        .catch(error => {
+          if (
+            error &&
+            error.response &&
+            error.response.status &&
+            error.response.status !== 400
+          ) {
+            setToError(true);
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/page/informationreview/InformationReview.js:110]: so we dont want to go to error page if error status is 400?

File: src/ecrc-frontend/src/components/page/informationreview/InformationReview.test.js [hunk 2]
```diff
@@ -230,6 +262,25 @@ describe("InformationReview Component", () => {
     expect(history.location.pathname).toEqual("/criminalrecordcheck/error");
   });
 
+  test("Sets error and redirects to error page when checkApplicantForPrevCRC fails with something other than 400", async () => {
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/page/informationreview/InformationReview.test.js:265]: i think we have a test case where it mocks a 400 return right? we should change the text on that test to say
```
Sets error and does not redirect to error page when checkApplicantForPrevCRC fails with 400
```

Comment (sdevalapurkar-bcgov) [reply src/ecrc-frontend/src/components/page/informationreview/InformationReview.test.js:265]: so it makes sense and is consistent with this test you added/the actual functionality

File: src/ecrc-frontend/src/components/page/informationreview/InformationReview.js (hunk 1 above)
Comment (adodge-bcg) [reply src/ecrc-frontend/src/components/page/informationreview/InformationReview.js:110]: Yes. 400 is what we get back from checkApplicantForPrevCRC when it does not find anything, so it is an expected error.

File: src/ecrc-frontend/src/components/page/informationreview/InformationReview.test.js (hunk 2 above)
Comment (adodge-bcg) [reply src/ecrc-frontend/src/components/page/informationreview/InformationReview.test.js:265]: All the tests that do not check share are currently running in the 400 state, only the ones that directly test share are in 200, and this one is in 500 to make sure it redirects when we get a non 400 error.

Comment (sdevalapurkar-bcgov) [reply src/ecrc-frontend/src/components/page/informationreview/InformationReview.test.js:265]: so can we add a share test case which is 400?

File: src/ecrc-frontend/src/components/page/informationreview/InformationReview.test.js [hunk 3]
```diff
@@ -81,15 +84,32 @@ describe("InformationReview Component", () => {
       authorities: ["Authorized"]
     });
 
-    const mock = new MockAdapter(axios);
-    const API_REQUEST_SHARE = "/ecrc/private/checkApplicantForPrevCRC";
+    mock.onPost(API_REQUEST_SHARE).reply(400, {
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/page/informationreview/InformationReview.test.js:87]: ok gotcha!

**Pull request merged**